docs/source/optimizers.rst

```diff
@@ -1,7 +1,88 @@
 .. _optimizers:
 
+************
 Optimization
-===============
+************
+
+Lightning offers two modes for managing the optimization process:
+
+- automatic optimization (AutoOpt)
+- manual optimization
+
+For the majority of research cases, **automatic optimization** will do the right thing for you and it is what
+most users should use.
+
+For advanced/expert users who want to do esoteric optimization schedules or techniques, use **manual optimization**.
+
+------
+
+Manual optimization
+===================
+For advanced research topics like reinforcement learning, sparse coding, or GAN research, it may be desirable
+to manually manage the optimization process. To do so, do the following:
+
+* Ignore the optimizer_idx argument
+* So we can scale the loss automatically for you use self.backward(loss) instead of loss.backward()
+
+.. code-block:: python
+
+    def training_step(self, batch, batch_idx, optimizer_idx):
+        # ignore optimizer_idx
+        (opt_g, opt_d) = self.optimizers()
+
+        # do anything you want
+        loss_a = ...
+
+        # use self.backward which will also handle scaling the loss when using amp
+        self.backward(loss_a, opt_g)
+        opt_g.step()
+        opt_g.zero_grad()
+
+        # do anything you want
+        loss_b = ...
+
+        # pass in any args that loss.backward() normally takes
+        self.backward(loss_b, opt_d, retain_graph=True)
+        self.backward(loss_b, opt_d, retain_graph=True)
+        loss_b.step()
+        loss_b.zero_grad()
+
+.. note:: This is only recommended for experts who need ultimate flexibility
+
+------
+
+Automatic optimization
+======================
+With Lightning 90% of users don't have to think about when to call .backward(), .step(), .zero_grad(), since
```

Comment:
```suggestion
With Lightning, most of users don't have to think about when to call .backward(), .step(), .zero_grad(), since
```
unless you're sure about the 90% :)